src/nuget/Fable.JsonConverter/Fable.JsonConverter.fs

```diff
@@ -36,15 +36,15 @@ type JsonConverter() =
         advance reader
         read 0 List.empty
 
-    let typeCache = Dictionary<Type, bool>()
+    let typeCache = ConcurrentDictionary<_, _>()
     override x.CanConvert(t) =
             match typeCache.TryGetValue(t) with
             | false, _ ->
                 let result = 
                     t.Name = "FSharpOption`1"
                     || FSharpType.IsTuple t
                     || (FSharpType.IsUnion t && t.Name <> "FSharpList`1")
-                typeCache.Add(t, result)
+                typeCache.TryAdd(t, result) |> ignore
                 result
             | _, result -> result
```

Comment:
Sorry, I forgot to tell you but `ConcurrentDictionary` has a `GetOrAdd` method which allows to do this in a more expressive way and I think is more performant (because you don't need to get the lock on the dictionary twice:

```fsharp
typeCache.GetOrAdd(t, fun t ->
    t.Name = "FSharpOption`1"
    || FSharpType.IsTuple t
    || (FSharpType.IsUnion t && t.Name <> "FSharpList`1"))
```